third_party/CMakeLists.txt

```diff
@@ -18,4 +18,5 @@ endif()
 add_subdirectory(re2)
 add_subdirectory(serd)
 add_subdirectory(utf8proc)
+add_subdirectory(yyjson)
```

Comment:
yyjson is only used by the json extension, so i suggest moving json library under extension/json/third-party/

Reply:
This may be a problem; extension/json/third-party gets picked up by clang-tidy, and yyjson.c does not conform to clang-tidy requirements at all.